components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -32,71 +34,457 @@ - (UIView *)createSideView {
 }
 
 - (void)testLeadingViewModeRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.leadingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.rightViewMode == textField.leadingViewMode,
                 @"The leading view mode should be equal to the right view mode.");
 }
 
 - (void)testTrailingViewModeLTR {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
   textField.trailingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.rightView == textField.trailingView,
                 @"The trailing view mode should be equal to the right view mode.");
 }
 
 - (void)testTrailingViewModeRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.trailingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.leftViewMode == textField.trailingViewMode,
                 @"The trailing view mode should be equal to left view mode.");
 }
 
+- (void)testAdjustTextAreaFrameWithParentClassTextAreaFrame {
+  // Given
+  CGRect textAreaFrame = CGRectMake(30, 50, 120, 20);
+  CGRect parentClassTextAreaFrame = CGRectMake(20, 30, 120, 50);
+
+  // When
+  CGRect finalTextAreaFrame =
+      [[[MDCBaseTextField alloc] init] adjustTextAreaFrame:textAreaFrame
+                              withParentClassTextAreaFrame:parentClassTextAreaFrame];
+
+  // Then
+  CGRect correctFinalTextAreaFrame = CGRectMake(30, 35, 120, 50);
+  XCTAssertEqualObjects(NSStringFromCGRect(correctFinalTextAreaFrame),
+                        NSStringFromCGRect(finalTextAreaFrame));
+}
+
+- (MDCBaseTextField *)createLTRTextFieldWithLeadingView:(BOOL)leadingView
+                                           trailingView:(BOOL)trailingView
+                                               viewMode:(UITextFieldViewMode)viewMode {
+  MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+  textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
+  if (leadingView) {
+    textField.leadingView = [self createSideView];
+  }
+  if (trailingView) {
+    textField.trailingView = [self createSideView];
+  }
+  textField.trailingViewMode = viewMode;
+  textField.leadingViewMode = viewMode;
+  UIWindow *window = [[UIWindow alloc] init];
+  [window addSubview:textField];
+  return textField;
+}
+
+- (void)forceLayoutOfView:(UIView *)view {
+  [view setNeedsLayout];
+  [view layoutIfNeeded];
+}
+
+- (BOOL)CGRect:(CGRect)CGRect1 isEqualToCGRect:(CGRect)CGRect2 {
```

Comment:
Did you find that `CGRectEqualToRect` wasn't working? If so, please document why here.

Reply:
I forgot it existed ahaha